src/herder/SurgePricingUtils.cpp

```diff
@@ -160,6 +160,40 @@ SurgePricingPriorityQueue::getMostTopTxsWithinLimits(
     SurgePricingPriorityQueue queue(
         /* isHighestPriority */ true, laneConfig,
         stellar::rand_uniform<size_t>(0, std::numeric_limits<size_t>::max()));
+
+    bool allFit = true;
+    auto res = queue.countTxsResources(txs, ledgerVersion);
+    auto totalResources = Resource::makeEmpty(res[0].size());
+
+    // First check if each individual lane fits within its own limit
+    for (int i = 0; i < static_cast<int>(queue.getNumLanes()); ++i)
+    {
+        // Check against individual lane limits
+        auto const& limit = queue.laneLimits(i);
+        if (anyGreater(res.at(i), limit))
+        {
+            allFit = false;
```

Comment:
Can we just `break` out of the loop at this point where we set `allFit = false`? The same question goes for the conditional following this one on line 180.